### PR TITLE
Fix Remote IP Address - NULL reference exception

### LINKED ISSUE #3276: Remote IP Address - NULL reference exception
# Bug Report

[OpenTelemetry.Instrumentation.AspNetCore](https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/src/OpenTelemetry.Instrumentation.AspNetCore) - v1.0.0-rc9.3

Runtime version: `net6.0`

## Symptom

`NullReferernceException` thrown when `HttpContext.Connection.RemoteIpAddress` is `null` due to a `.toString()` call in [`HttpInListener.AddGrpcAttributes`](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInListener.cs#L361)

**What is the expected behavior?**

The `AttributeNetPeerIp` should not be added to the activity when the `RemoteIpAddress` property is NULL

**What is the actual behavior?**

Unhandled `NullReferenceException`

## Reproduce

See [RemoteIpAddress]( https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.http.connectioninfo.remoteipaddress?view=aspnetcore-6.0) documentation indicating the property can be null.

### PR DESCRIPTION
**Pull request by nayang7**
fix Remote IP Address - NULL reference exception

Fixes #3276.

## Changes

Please provide a brief description of the changes here.

For significant contributions please make sure you have completed the following items:

* [ ] Appropriate `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

**Comment by reyang** [comment]
@nayang7 thanks for your contribution! Would you clear the CLA and update the changelog?

**Comment by nayang7** [comment]
> @nayang7 thanks for your contribution! Would you clear the CLA and update the changelog? (and ideally add a test case)

@reyang I have updated the changelog and CLA was resolved, btw, I have to say current didn't have test file about '(OpenTelemetry.Instrumentation.AspNetCore/Implementation)/HttpInListener.cs' so it's really hard to test Http here.

File: src/OpenTelemetry.Instrumentation.AspNetCore/CHANGELOG.md
Comment (reyang) [inline src/OpenTelemetry.Instrumentation.AspNetCore/CHANGELOG.md:6]:
```suggestion
  [#3481](https://github.com/open-telemetry/opentelemetry-dotnet/pull/3481))
```

**Review by reyang — APPROVED** [review]

**Review by cijothomas — APPROVED** [review]

**Review by Kielek — APPROVED** [review]

**Comment by nayang7** [comment]
@reyang  hey reyang, it seems 'First-time contributors need a maintainer to approve running workflows.'

**Comment by reyang** [comment]
> @reyang hey reyang, it seems 'First-time contributors need a maintainer to approve running workflows.'

I just approved the CI workflows.

**Comment by nayang7** [comment]
@reyang hey reyang, sorry it's my fault, I didn't notice all the format error msg, I just updated fixing 'error SA1513: Closing brace should be followed by blank line'. Thanks your approve again.

**Comment by nayang7** [comment]
@reyang hey reyang, the format really confused me, need your approval again for 'SA1028 Code Should Not Contain Trailing Whitespaces', thanks.

**Comment by nayang7** [comment]
@reyang @cijothomas Just now it seems the checking all passed and with coverage report. Since it lasts several days, just Update branch, then needs approval again. Thanks

**Comment by nayang7** [comment]
@reyang  @cijothomas  it seems API Compatibility build-test failed after I update branch

'D:\a\opentelemetry-dotnet\opentelemetry-dotnet\build\Common.prod.props(20,5): error MSB3073: The command "powershell -ExecutionPolicy Unrestricted -File "D:\a\opentelemetry-dotnet\opentelemetry-dotnet\build\PreBuild.ps1" -package OpenTelemetry -version "1.3.0"" exited with code 1. [D:\a\opentelemetry-dotnet\opentelemetry-dotnet\src\OpenTelemetry\OpenTelemetry.csproj]
    0 Warning(s)
    1 Error(s)'

**Comment by cijothomas** [comment]
> @reyang @cijothomas it seems API Compatibility build-test failed after I update branch
> 
> 'D:\a\opentelemetry-dotnet\opentelemetry-dotnet\build\Common.prod.props(20,5): error MSB3073: The command "powershell -ExecutionPolicy Unrestricted -File "D:\a\opentelemetry-dotnet\opentelemetry-dotnet\build\PreBuild.ps1" -package OpenTelemetry -version "1.3.0"" exited with code 1. [D:\a\opentelemetry-dotnet\opentelemetry-dotnet\src\OpenTelemetry\OpenTelemetry.csproj] 0 Warning(s) 1 Error(s)'

That is a bit flaky test. However it is not mandatory check, so won't block this PR. 
Could you rebase so that we can merge?

**Comment by nayang7** [comment]
> 

@cijothomas thanks, I have rebased

**Pull request merged**
